tests/reftests/clean.test

```diff
@@ -0,0 +1,91 @@
+N0REP0
+### <pkg:things.1>
+opam-version: "2.0"
+build : [ "touch" "first" "second" ]
+install: [
+  [ "test" "!" "-f" "%{bin}%/first" ]
+  [ "cp" "first" "%{bin}%/first" ]
+  [ "test" "!" "-f" "%{lib}%/things/second" ]
+  [ "mkdir" "%{lib}%/things/" ]
+  [ "cp" "second" "%{lib}%/things/second" ]
+]
+### OPAMPRECISETRACKING=1 OPAMYES=1
+### opam update
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam switch create clean --empty
+### opam install things
+The following actions will be performed:
+  - install things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed things.1
+Done.
+### opam show --list-files things
+${BASEDIR}/OPAM/clean/bin/first
+${BASEDIR}/OPAM/clean/lib/things
+${BASEDIR}/OPAM/clean/lib/things/second
+
+### opam remove things
+The following actions will be performed:
+  - remove things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   things.1
+Done.
+### opam install things
+The following actions will be performed:
+  - install things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed things.1
+Done.
+### opam remove things --fake
+The following actions will be faked:
+  - remove things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+Done.
+### opam install things |  grep -v '^#.*'
+The following actions will be performed:
+  - install things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+[ERROR] The installation of things failed at "test ! -f ${BASEDIR}/OPAM/clean/bin/first".
+
+
+
+
+<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
++- The following actions failed
+| - install things 1
++- 
+- No changes have been performed
+# Return code 31 #
+### opam clean --untracked --dry-run
+rm -rf ${BASEDIR}/OPAM/clean/bin/first
+rm -rf ${BASEDIR}/OPAM/clean/bin
+rm -rf ${BASEDIR}/OPAM/clean/lib/things/second
+rm -rf ${BASEDIR}/OPAM/clean/lib/things
+Cleaning up switch clean
+rm -rf "${BASEDIR}/OPAM/clean/.opam-switch/backup"/*
+rm -rf "${BASEDIR}/OPAM/clean/.opam-switch/build"/*
+rm -rf "${BASEDIR}/OPAM/clean/.opam-switch/remove"/*
+rm -rf "${BASEDIR}/OPAM/clean/.opam-switch/extra-files-cache"/*
+### opam clean --untracked
+Non empty directory ${BASEDIR}/OPAM/clean/bin containing:
+  - ${BASEDIR}/OPAM/clean/bin/first
+Remove files? [Y/n] y
+Non empty directory ${BASEDIR}/OPAM/clean/lib/things containing:
+  - ${BASEDIR}/OPAM/clean/lib/things/second
+Remove directory? [Y/n] y
+Cleaning up switch clean
+### opam install things
+The following actions will be performed:
+  - install things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed things.1
+Done.
```

Comment:
Could you add something like:
```
### <modify an installed file>
### opam clean --untracked
<see if it’s actually removed>
### opam clean --untracked
<see if nothing is actually done>
```

Reply:
with an installed or not installed package ?

Reply:
why not both, just in case

Reply:
ok